Translate: handling events docs (#77)
reactjs/es.reactjs.org: content/docs/handling-events.md

---
id: handling-events
title: Manejando Eventos
permalink: docs/handling-events.html
prev: state-and-lifecycle.html
next: conditional-rendering.html
redirect_from:
  - "docs/events-ko-KR.html"
---

> Manejar eventos con elementos de React es muy similar a manejar eventos con elementos del Dom.. Solo hay una diferencias de sintaxis:

Reviewer: I suggest
```suggestion
Manejar eventos con elementos de React es muy similar a manejar eventos en elementos del DOM. Hay algunas diferencias de sintaxis:
```

> Manejar eventos con elementos de React es muy similar a la manera en cómo se manejan eventos con elementos del DOM. Solo hay una diferencias de sintaxis:

Reviewer: I suggest `...es muy similar a manejar eventos con elementos del Dom.`

* Los eventos de React se nombran usando camelCase, en vez de minúsculas.

> * En JSX se pasa la función como el manejador del evento, en vez de una cadena de texto.

Reviewer: Could be:
```suggestion
* Con JSX pasas una función como el manejador del evento, en vez de un string.
```
Here are some words that you should consider to not translate, as string: https://github.com/reactjs/es.reactjs.org/blob/master/TRANSLATION.md

Por ejemplo, el HTML:

```html
<button onclick="activateLasers()">
  Activate Lasers
</button>
```

En React es algo diferente:

```js{1}
<button onClick={activateLasers}>
  Activate Lasers
</button>
```

> Otra diferencia es que en React no puedes retornar `false` para prevenir el comportamiento por defecto. Debes, explícitamente, llamar `preventDefault`. Por ejemplo, en un HTML plano, para prevenir el comportamiento por defecto de un hipervínculo de abrir una nueva página, puedes escribir:

Reviewer: What about `enlace` for `link` or just `link`

}
```

> Aquí, `e` es un evento sintético. React define estos eventos sintéticos acorde a las [especificaciones W3C](https://www.w3.org/TR/DOM-Level-3-Events/), entoncés no debes preocuparte por la compatibilidad a tráves de los navegadores. Mira la guía de referencia [`SyntheticEvent`](/docs/events.html) para aprender más.

Reviewer: ```suggestion
Aquí, `e` es un evento sintético. React define estos eventos sintéticos acorde a las [especificaciones W3C](https://www.w3.org/TR/DOM-Level-3-Events/), entonces no debes preocuparte por la compatibilidad a tráves de los navegadores. Mira la guía de referencia [`SyntheticEvent`](/docs/events.html) para aprender más.
```

> Aquí, `e` es un evento sintético. React define estos eventos sintéticos acorde a las [especificaciones W3C](https://www.w3.org/TR/DOM-Level-3-Events/), entonces no debes preocuparte por la compatibilidad a tráves de los navegadores. Mira la guía de referencia [`SyntheticEvent`](/docs/events.html) para aprender más.

Reviewer: What do you think about `deberías` instead of `debes`

Author: I think `deberías` sounds like some times you won't need to worry, but in some cases you will. In that case it would be `... you shouldn't need to worry...`. LMK what do you think about this one

> Cuando estás utilizando React, generalmente, no debes llamar `addEventListener` para agregar manejadores de eventos a un elemento del DOM después de que este es creado. Por el contrario, solo debes proveer un manejador de eventos cuando el elemento es inicialmente renderizado.

Reviewer: I think listeners aren't the same of handlers, so IMO you could change `escuchadores or receptores` instead of `manejadores`

> Cuando defines un componente usando [clases de ES6](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes), un patrón muy común es que los manejadores de eventos sean un método de la clase. Por ejemplo, este componente `Toggle` renderiza un botón que permite al usuario cambiar el estado entre "PRENDIDO" y "APAGADO":

Reviewer 2: Let's reference the Spanish MDN docs :)

```suggestion
Cuando defines un componente usando [clases de ES6](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes), un patrón muy común es que los manejadores de eventos sean un método de la clase. Por ejemplo, este componente `Toggle` renderiza un botón que permite al usuario cambiar el estado entre "PRENDIDO" y "APAGADO":
```

> Cuando defines un componente usando [clases de ES6](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes), un patrón muy común es que los manejadores de eventos sean un método de la clase. Por ejemplo, este componente `Toggle` renderiza un botón que permite al usuario cambiar el estado entre "PRENDIDO" y "APAGADO":

Reviewer: IMO encendido sounds better than prendido. What do you think?
```suggestion
Cuando defines un componente usando una [clase de ES6](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Classes), un patrón muy común es que los manejadores de eventos sean un método de la clase. Por ejemplo, este componente `Toggle` renderiza un botón que permite al usuario cambiar el estado entre "ENCENDIDO" y "APAGADO":
```

```js{6,7,10-14,18}
class Toggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = {isToggleOn: true};

>     // Esta ligadura es necesaria para hacer que `this` funcione en el callback

Reviewer: What do you think about vínculo or enlace instead of ligadura?

);
```

[**Pruébalo en CodePen**](http://codepen.io/gaearon/pen/xEmzGg?editors=0010)

Tienes que tener mucho cuidado en cuanto al significado de `this` en los callbacks de JSX. En JavaScript, los métodos de clase no están [ligados](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind) por defecto. Si olvidas ligar `this.handleClick` y lo pasas a `onClick`, `this` sera `undefined` cuando se llame la función.

> Esto no es un comportamiento especifico de React; esto hace parte de [como funcionan las funciones JavaScript](https://www.smashingmagazine.com/2014/01/understanding-javascript-function-prototype-bind/). Generalmente, si refieres un método sin usar `()` después de este, tal como `onClick={this.handleClick}`, deberías ligar ese método.

Reviewer: Maybe you could change `como funcionan las funciones`  by `como operan las funciones` to not be repetitve. 

Also IMO sounds better 

> esto es parte de 

than

> esto hace parte de

Author: Don't know about `esto es parte de`. I'm open to hearing opinions

}
```

Esta sintaxis está habilitada por defecto en [Create React App](https://github.com/facebookincubator/create-react-app).

> :

Reviewer: ```suggestion

```

> Si no estas usando la sintaxis de campos publicos de clases, puedes usar [funciones flecha](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) en el callback:

Reviewer: ```suggestion
Si no estas usando la sintaxis de campos publicos de clases, puedes usar una [función flecha](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) en el callback:
```

}
```

> El problema con esta sintaxis es que un callback diferente es creado cada vez que `LogginButton` es renderizado. En la mayoría de los casos, esto está bien. Sin embargo, si este callback se pasa como una propiedad a componentes más bajos, estos componentes podría renderizarse nuevamente. Nosotros, generalmente, recomendamos ligar en el constructor o usar la sintaxis de campos de clases, para evitar esta clase de problemas de rendimiento.

Reviewer: ```suggestion
El problema con esta sintaxis es que un callback diferente es creado cada vez que `LogginButton` es renderizado. En la mayoría de los casos, esto está bien. Sin embargo, si este callback se pasa como una propiedad a componentes más bajos, estos componentes podrían renderizarse nuevamente. Generalmente, recomendamos ligar en el constructor o usar la sintaxis de campos de clases, para evitar esta clase de problemas de rendimiento.
```

> ## Pasando Argumentos a un Manejador de Eventos {#passing-arguments-to-event-handlers}

Reviewer: I suggest
```suggestion
## Pasando Argumentos a Manejadores de Eventos {#passing-arguments-to-event-handlers}
```

> Dentro de un bucle es muy común querer pasar un parámetro extra a un manejador de eventos. Por ejemplo, si `id` es el ID de una fila, cualquiera de los códigos a continuación podría funcionar:

Reviewer: I suggest
```suggestion
Dentro de un bucle es muy común querer pasar un parámetro extra a un manejador de eventos. Por ejemplo, si `id` es el ID de una fila, cualquiera de los códigos siguientes podría funcionar:
```

Author: IMHO, `a continuación` sounds better. 

Maybe `cualquiera de los siguientes códigos...`

Hearing opinions.
